source/common.c

```diff
@@ -28,34 +33,114 @@
 #    pragma warning(disable : 4100)
 #endif
 
-static void *s_default_malloc(struct aws_allocator *allocator, size_t size) {
+#ifdef AWS_DEFAULT_ALLOCATOR_LEAKCHECK
+struct aws_allocation {
+    const char *file;
+    int line;
+    size_t size;
+
+    struct aws_allocation *next;
```

Comment:
We have an intrusive linked-list already, shouldn't we be using those helper functions?